components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/MobileNumberVerificationHandler.java

```diff
@@ -90,6 +92,25 @@ public void handleEvent(Event event) throws IdentityEventException {
                 (String) eventProperties.get(IdentityEventConstants.EventProperty.TENANT_DOMAIN),
                 userStoreManager.getRealmConfiguration().
                         getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME));
+
+        if (IdentityEventConstants.Event.PRE_DELETE_USER_CLAIM.equals(eventName)) {
```

Comment:
Shall we move this code after this?
```
Map<String, String> claims = (Map<String, String>) eventProperties.get(IdentityEventConstants.EventProperty
                .USER_CLAIMS);
        if (claims == null) {
            claims = new HashMap<>();
        }
```
Since here, we already resolve claims from event properties.